File: modules/sql/app/model/querycorrectors/ColumnMatch.java
package model.querycorrectors;

import model.matching.Match;
import model.matching.MatchType;
import play.twirl.api.Html;

public class ColumnMatch extends Match<ColumnWrapper> {
  
  protected boolean hasAlias;

  protected boolean restMatched;

  public ColumnMatch(ColumnWrapper theArg1, ColumnWrapper theArg2) {
    super(theArg1, theArg2);
  }
  
  public boolean colNamesMatched() {
    return matchType == MatchType.SUCCESSFUL_MATCH || matchType == MatchType.UNSUCCESSFUL_MATCH;
  }
  
  @Override
  public Html describe() {
    return views.html.resultTemplates.columnResult.render(this);
  }
  
  public String getFirstColName() {
    return userArg.getColName();
  }
  
  public String getFirstRest() {
    return sampleArg.getRest();
  }
  
  public String getSecondColName() {
    return sampleArg.getColName();
  }
  
  public String getSecondRest() {
    return sampleArg.getRest();
  }
  
  public boolean hasAlias() {
    return hasAlias;
  }
  
  public boolean restMatched() {
    return restMatched;
  }
  
  @Override
  protected MatchType analyze(ColumnWrapper theArg1, ColumnWrapper theArg2) {
    // TODO Auto-generated method stub
    return null;
  }
  
}

File: modules/sql/app/model/querycorrectors/ColumnMatch.scala
package model.querycorrectors;

import model.CorrectionException;
import model.matching.ScalaMatch;

import model.matching.MatchType;
import play.Logger;
import play.twirl.api.Html;

case class ColumnMatch(userCol: Option[ColumnWrapper], sampleCol: Option[ColumnWrapper]) extends ScalaMatch[ColumnWrapper](userCol, sampleCol) {

  val hasAlias: Boolean = userArg.get.hasAlias || sampleCol.get.hasAlias

  val restMatched: Boolean = false

  val colNamesMatched = matchType == MatchType.SUCCESSFUL_MATCH || matchType == MatchType.UNSUCCESSFUL_MATCH

  val firstColName = userArg.get.getColName;

  val firstRest = userArg.get.getRest;

  val secondColName = sampleArg.get.getColName;

  val secondRest = sampleArg.get.getRest;

  override def analyze(userArg: ColumnWrapper, sampleArg: ColumnWrapper) = userArg.doMatch(sampleArg)

  override def describe =
    <tr class={ getBSClass }>
  		<td><span class={ matchType.toString }></span></td>
  		<td>
				<span class={ if (colNamesMatched) "text-success" else "text-danger" }>{ firstColName }</span>
    		<span class={ if (restMatched) "text-success" else "text-danger" }>{ firstRest }</span>
  		</td>
 		  <td>
   	 		<span class={ if (colNamesMatched) "text-success" else "text-danger" }>{ secondColName }</span>
    		<span class={ if (restMatched) "text-success" else "text-danger" }>{ secondRest }</span>
  		</td>
    	<td>{ explanation }</td>
		</tr>

}
